internal/service/alarms/internal/dictionary/dictionary.go

```diff
@@ -0,0 +1,187 @@
+package dictionary
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"sync"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift-kni/oran-o2ims/internal/clients"
+)
+
+const (
+	managedClusterVersionLabel = "openshiftVersion-major-minor"
+	localClusterLabel          = "local-cluster"
+
+	resourceTypeCluster = "Cluster"
+)
+
+type AlarmDictionary struct {
+	Client client.Client
+
+	RulesMap map[int][]*monitoringv1.PrometheusRule
+}
+
+func New(client client.Client) *AlarmDictionary {
+	return &AlarmDictionary{
+		Client:   client,
+		RulesMap: make(map[int][]*monitoringv1.PrometheusRule),
+	}
+}
+
+// Load loads the alarm dictionary
+func (r *AlarmDictionary) Load(ctx context.Context) {
+	// TODO: List Resource Type from Resource Server
+	resourceTypeList := getResourceTypes()
+
+	type result struct {
+		resourceTypeID int
+		rules          []*monitoringv1.PrometheusRule
+		err            error
+	}
+
+	// If there are many resource types, we can use a semaphore to limit the number of concurrent goroutines
```

Comment:
how are we limiting the concurrency? Shouldnt there be there something `semaphore := make(chan struct{}, maxConcurrency)`...which potentially makes wg redundant? 

Unless meant to just synchronize (in that case code is good but comment is not i think)?

Reply:
That would be the case if we have a large number of resource types, but I don't expect that to be the case. The comment is more for completeness.

I see that the comment might be confusing, the idea is not to replace the wg with a buffered channel, but limit one specific section inside the routines.  I will delete the comment for now.

Reply:
Done